bin/prover-client/src/proving_ops/btc_ops.rs

```diff
@@ -36,29 +36,42 @@ impl BtcOperations {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct L1BlockscanInput {
+    pub blocks: Vec<Block>,
+    pub rollup_params: RollupParams,
+}
+
 #[async_trait]
 impl ProvingOperations for BtcOperations {
-    type Input = (Block, RollupParams);
-    type Params = u64; // params is the block height
+    type Input = L1BlockscanInput;
+    type Params = (u64, u64);
```

Comment:
I left this in a comment on another PR but we really should be using blkids here instead of block heights so that it's always consistent and eliminates any risk of accidentally using unexpected blocks.